src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -620,7 +620,14 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
 {
    # read as YAML
    opts <- .rs.fromYAML(opt)
-   
+
+   # knitr version 1.44 introduced a function `knitr:::dot_names`
+   # which converts dashes to dots in chunk options
+   if (packageVersion("knitr") >= "1.44") {
+      opts <- knitr:::dot_names(opts)
```

Comment:
Since we're calling an unexported function which might change (or not exist) in a future release, it might actually be better to do something like this:

```
knitr <- asNamespace("knitr")
if (is.function(knitr$dot_names)) {
  opts <- knitr$dot_names(opts)
}
```

I realize I made the initial recommendation to call it directly but I didn't realize it was an unexported function as first.